### PR TITLE
Add new unit and integration tests for activity input and expression evaluation

### PR DESCRIPTION
**Pull request by sfmskywalker**
- Introduced comprehensive test suites covering activity input evaluation, expression handling, and fault scenarios.
- Added unit tests for `ExpressionDescriptorRegistry`, `ExpressionEvaluator`, and activity execution context extensions.
- Added integration tests: `CustomInputEvaluatorTests`, `InputEvaluationErrorTests`, and `InputPropertyEvaluationTests`.
- Enhanced code coverage for edge cases and async evaluation logic.

- - -
This change is [Reviewable](https://reviewable.io/reviews/elsa-workflows/elsa-core/6990)

### CODE REVIEW

File: src/common/Elsa.Testing.Shared.Integration/RunActivityExtensions.cs [hunk 1]
```diff
@@ -42,4 +43,9 @@ public static async Task<RunWorkflowResult> RunActivityAsync(this IServiceProvid
         var result = await workflowRunner.RunAsync(activity, options, cancellationToken);
         return result;
     }
+    
+    public static async Task<RunWorkflowResult> RunWorkflowAsync(this IServiceProvider services, Workflow workflow, RunWorkflowOptions options, CancellationToken cancellationToken = default)
+    {
+        return await services.RunActivityAsync(workflow, options, cancellationToken);
+    }
```
[the remark this answers is not in the document]
Comment (lukhipolito-nexxbiz) [reply src/common/Elsa.Testing.Shared.Integration/RunActivityExtensions.cs]: seems like a valid comment

[the remark this answers is not in the document]
Comment (sfmskywalker) [reply src/common/Elsa.Testing.Shared.Integration/RunActivityExtensions.cs]: Yes, looks like an outdated comment though - XML comments are there. Resolving.
